lakefile.lean

```diff
@@ -46,7 +46,9 @@ def widgetJsAllTarget (pkg : Package) (isDev : Bool) : FetchM (Job (Array FilePa
   pkg.afterBuildCacheAsync $ deps.mapM fun depInfo => do
     let traceFile := pkg.buildDir / "js" / "lake.trace"
     let _ ← buildUnlessUpToDate? traceFile (← getTrace) traceFile do
-       /- HACK: Ensure that NPM modules are installed before building TypeScript,
+      if let some msg := get_config? errorOnBuild then
+        error msg
```

Comment:
```suggestion
      -- Downstream projects can require ProofWidgets using the syntax ??????
      -- in order to force ProofWidgets to fail with a custom error if it attempts to build,
      -- in order to prevent unactionable errors messages about `npm` being displayed to users.
      if let some msg := get_config? errorOnBuild then
        error msg
```

Reply:
@tydeu, could you fill out that comment?

Reply:
@kim-em Does my edit look good?

Reply:
I think this belongs to the README rather than here, so took the liberty of adding it. Btw, I'm assuming the syntax is actually ``insert `errorOnBuild "<my message>"``, with no backtick before `"<my message>"`?

Reply:
Looks good to me! Let's merge so Mathlib can take advantage of it.